Added an install script (linux only at the moment)

TheBearodactyl opened the pull request:
Obviously, the script can be improved on, but it'll get everything done!
Also, it works with both non-flatpak and Flatpak installations so that nobody
is left out! I'm currently working on an install script for Windows as well!
I've tested it out on my machine, and it works pretty well!

TheBearodactyl committed:
Create install.sh

TheBearodactyl committed:
V1

TheBearodactyl committed:
Update install.sh

TheBearodactyl committed:
Update README.md

TheBearodactyl commented:
Why tf did it close the pull request.

sydrinea commented on install.sh:
```diff
@@ -0,0 +1,112 @@
+
+# MAKE DIRECTORY SO THAT YOUR HOME DIRECTORY DOESN'T GET DIRTY
+
+mkdir -p junk-drawer
```
Might consider using `/tmp` instead. Also, this isn't cleaned up after the script finishes.

TheBearodactyl replied:
Oh lol, i didn't use /tmp because some people have a tendency to use a seperate partition for that directory or use cron jobs to clear that out (which could interfere with the installation)

sydrinea commented on install.sh:
```diff
@@ -0,0 +1,112 @@
+
+# Install git and wget
+
+pacman -S wget git --needed
```
This will fail on distributions that aren't Arch-based. Could fall back to `curl`, and if that also fails, quit early and prompt the user to install.

TheBearodactyl replied:
lemme add a thingy that'll ask for their distro

TheBearodactyl replied:
> lemme add a thingy that'll ask for their distro

done

sydrinea commented on install.sh:
```diff
@@ -0,0 +1,112 @@
+git clone https://github.com/catppuccin/steam
+cd steam
+cp -r $FLAVOR $METRO
+clear
```
it would be more useful to not clear the output, in case there's an error.

TheBearodactyl replied:
oki, i'll edit that lol

TheBearodactyl replied:
I fixed both things you mentioned!

TheBearodactyl replied:
I just removed that part lol

TheBearodactyl committed:
Added distro-specific commands for installing wget and git

TheBearodactyl committed:
Update install.sh

TheBearodactyl committed:
Update install.sh

NOTICE:
The reason I have it make a new folder instead of using /tmp is because:
1) Some people use a seperate partition for /tmp and this makes it harder to use
2) Some people like to use cron jobs to automatically clear /tmp and that can interfere with the installation process

Oman395 commented on install.sh:
```diff
@@ -0,0 +1,165 @@
+
+# Install git and wget
+
+echo "What distro do you use?"
```
We could do this automatically, I.E
```sh
cat /etc/issue | grep -Er (insert regex here, I'm not sure the format of /etc/issue lol)

TheBearodactyl replied:
Ok, so i found a way to find out the users distro automatically.
```
export DISTRO=$(echo "$(lsb_release -is)")
```

Does that look good?

Oman395 commented on install.sh:
```diff
@@ -0,0 +1,165 @@
+  ;;
+esac
+
+# Download, unzip, and execute SPF (formerly Steam Friends Patcher)
```
SFP not SPF

TheBearodactyl replied:
LOL I'M STUPID. I was trying to download sunscreen :skull_and_crossbones:

Oman395 commented on install.sh:
```diff
@@ -0,0 +1,165 @@
+# then makes a folder in the .steam/steam/skins directory called 'metro' to unzip metro to.
+# Also makes the skins folder if it doesn't exist
+
+read -p "Where is your steam installation? Copy the path (e.g. /home/thebearodactyluwu/.steam) and paste it here: " steaminstall
```
It might be better to check if $HOME/.steam exists, and if not ask this question, maybe saying "Steam directory not found. please copy and paste it here: "

Oman395 commented on install.sh:
```diff
@@ -0,0 +1,165 @@
+
+# Removes the junk-drawer folder from earlier
+
+cd /home/$USER
```
$HOME

Oman395 commented on install.sh:
```diff
@@ -0,0 +1,165 @@
+rm -rf junk-drawer
+
+# Little easter egg for those who don't mind waiting about 15 seconds a lot 
+echo "Why're you still here? You"
```
This is funny, but it would be funnier if it was a stanley parable style dialog that lasts several minutes

TheBearodactyl replied:
OH MY GOD YES

Oman395 reviewed (changes requested):
looks good, but there are typos and there's a couple things that could be more automated

Oman395 commented:
Currently making some changes for ya to take a look at

Oman395 commented:
Here it is! Fixed up some typos, replaced /home/$USER with $HOME, made the sudo access thing so that it's only if equal to 0, because as it was it would only allow you to run as root, removed the distro thing and replaced it with || exit on the packages that might not be installed; the error messages should be enough to tell the user what's wrong, and it's not exactly best practice to force install dependencies without the user giving feedback. I also removed the running steam thing, because not everyone installs steam with flatpak-- it could probably be replaced with just "steam". I also removed the clear command; it didn't really do anything, and was mildly annoying.
```sh
#!/bin/bash

# CHECKS FOR SUDO ACCESS (added because of those who have seperate partitions for /home and /root) 

if [ "$EUID" -eq 0 ]
  then echo "Please don't run as root" && exit
  exit
fi

# MAKE DIRECTORY SO THAT YOUR HOME DIRECTORY DOESN'T GET DIRTY

cd "$HOME"
mkdir -p junk-drawer
cd junk-drawer

# Download, unzip, and execute SFP (formerly Steam Friends Patcher)

wget https://github.com/PhantomGamers/SFP/releases/download/0.0.14/SFP_UI-SelfContained-linux-x64.zip || exit
unzip SFP_UI-SelfContained-linux-x64.zip || exit
echo "SFP Downloaded, please press \"patch\" when the window opens"
chmod +x SFP_UI
./SFP_UI

# Download the original Steam for Metro (Will create a directory for it so that things stay clean) 

wget https://github.com/minischetti/metro-for-steam/archive/refs/tags/v4.4.zip
mkdir ./metro && mv v4.4.zip ./metro/ && cd ./metro

# Unzip Metro and remove the ZIP file

unzip ./v4.4.zip
rm ./v4.4.zip

# Ask for the path to the users .steam file so that it can be used with Flatpak installations,
# then makes a folder in the .steam/steam/skins directory called 'metro' to unzip metro to.
# Also makes the skins folder if it doesn't exist

read -pr "Where is your steam installation? Copy the path (e.g. /home/thebearodactyluwu/.steam) and paste it here: " steaminstall
mkdir -p "$steaminstall/steam/skins/metro"

# Sets the METRO variable to the previously created metro folder, then copies the 
# extracted contents of metro to said metro folder

METRO="$steaminstall/steam/skins/metro"
cp -r ./* "$steaminstall/steam/skins/metro"
rm -rf ./*

# Downloads the latest release of the Metro Patch

wget https://github.com/redsigma/UPMetroSkin/archive/refs/tags/9.1.44.zip
unzip 9.1.44.zip
cp -r "./9.1.44/'Unofficial 4.x Patch'/'Main Files [Install First]'/*" "$METRO"
printf 'What is your flavour?\n 1) Frappe\n 2) Latte\n 3) Macchiato\n 4) Mocha\n'

# Asks the user which flavour they'd like to install
# FLAVOR is the theme folder
# FLAVOUR is literally just the flavor name so that the little echo scripts at the end look good

case $FLAVOR in 
[1])
	FLAVOR='themes/frappe'
  FLAVOUR='Frappe'
	;;

[2])
	FLAVOR='themes/latte'
  FLAVOUR='Latte'
	;;

[3])
  FLAVOR='themes/macchiato'
  FLAVOUR='Macchiato'
  ;;

[4])
	FLAVOR='themes/mocha'
  FLAVOUR='Mocha'
	;;
esac
git clone https://github.com/catppuccin/steam || exit
cd steam
cp -r "$FLAVOR" "$METRO"

# Gives the user instructions to complete the installation

echo "Steps to install theme:"
echo " 1) Open Steam and go into settings"
echo " 2) Change the skin to metro"
echo " 3) Steam will prompt you to restart!"
echo "    Just press OK, and presto!"
echo "    You have installed: "
echo
echo "    Catppuccin " "$FLAVOUR"
echo
echo "    On Steam!"

# Removes the junk-drawer folder from earlier

rm -rf "$HOME/junk-drawer"

sleep 15

# Little easter egg for those who don't mind waiting about 15 seconds a lot 
echo "Why're you still here? You"
echo "should be playing ULTRAKILL"
echo "or something!"

exit
```

Oman395 commented:
Looks like i missed some issues, one sec

Oman395 commented:
...ok, I just managed to nearly clean out my $HOME, I'm gonna stop working on this while I fix it

find literally any way to avoid using -f for rm, please, because I think that if the steam path is empty or sum it runs rm -rf $HOME

TheBearodactyl commented:
> ...ok, I just managed to nearly clean out my $HOME, I'm gonna stop working on this while I fix it
> 
> find literally any way to avoid using -f for rm, please, because I think that if the steam path is empty or sum it runs rm -rf $HOME

OH. YEAH, LEMME FIX THAT

TheBearodactyl commented:
> Collaborator



> ...ok, I just managed to nearly clean out my $HOME, I'm gonna stop working on this while I fix it
> 
> find literally any way to avoid using -f for rm, please, because I think that if the steam path is empty or sum it runs rm -rf $HOME

Ok, so imma close the pull request until i've fixed all those bugs, and i'll re-open it once i'm done!

Oman395 commented:
Just open a draft pr, so we can see progress and lyk if there's anything wrong

TheBearodactyl commented:
> Just open a draft pr, so we can see progress and lyk if there's anything wrong

Oh lol, good idea
